packages/contracts-bedrock/test/L1/SystemConfig.t.sol

```diff
@@ -220,99 +220,6 @@ contract SystemConfig_Initialize_Test is SystemConfig_TestInit {
-
-        // Verify that the l2ChainId was updated.
-        assertEq(systemConfig.l2ChainId(), 1234);
-
-        // Verify that the dispute game factory address was cleared.
-        assertEq(vm.load(address(systemConfig), disputeGameFactorySlot), bytes32(0));
-    }
-
-    /// @notice Tests that the upgrade() function reverts if called a second time.
-    function test_upgrade_upgradeTwice_reverts() external {
-        // Get the slot for _initialized.
-        StorageSlot memory slot = ForgeArtifacts.getSlot("SystemConfig", "_initialized");
-
-        // Set the initialized slot to 0.
-        vm.store(address(systemConfig), bytes32(slot.slot), bytes32(0));
-
-        // Trigger first upgrade.
-        vm.prank(address(systemConfig.proxyAdmin()));
-        systemConfig.upgrade(1234, ISuperchainConfig(address(0xdeadbeef)));
-
-        // Try to trigger second upgrade.
-        vm.prank(address(systemConfig.proxyAdmin()));
-        vm.expectRevert("Initializable: contract is already initialized");
-        systemConfig.upgrade(1234, ISuperchainConfig(address(0xdeadbeef)));
-    }
-
-    /// @notice Tests that the upgrade() function reverts if called after initialization.
-    function test_upgrade_afterInitialization_reverts() external {
-        // Get the slot for _initialized.
-        StorageSlot memory slot = ForgeArtifacts.getSlot("SystemConfig", "_initialized");
-
-        // Slot value should be set to 2 (already initialized).
-        bytes32 initializedSlotBefore = vm.load(address(systemConfig), bytes32(slot.slot));
-        assertEq(initializedSlotBefore, bytes32(uint256(2)));
-
-        // l2ChainId should be non-zero.
-        assertNotEq(systemConfig.l2ChainId(), 0);
-
-        // Try to trigger upgrade().
-        vm.expectRevert("Initializable: contract is already initialized");
-        systemConfig.upgrade(1234, ISuperchainConfig(address(0xdeadbeef)));
-    }
-
-    /// @notice Tests that the upgrade() function reverts if called by a non-proxy admin or owner.
-    /// @param _sender The address of the sender to test.
-    function testFuzz_upgrade_notProxyAdminOrProxyAdminOwner_reverts(address _sender) public {
-        // Prank as the not ProxyAdmin or ProxyAdmin owner.
-        vm.assume(_sender != address(systemConfig.proxyAdmin()) && _sender != systemConfig.proxyAdminOwner());
-
-        // Get the slot for _initialized.
-        StorageSlot memory slot = ForgeArtifacts.getSlot("SystemConfig", "_initialized");
-
-        // Set the initialized slot to 0.
-        vm.store(address(systemConfig), bytes32(slot.slot), bytes32(0));
-
-        // Expect the revert with `ProxyAdminOwnedBase_NotProxyAdminOrProxyAdminOwner` selector.
-        vm.expectRevert(IProxyAdminOwnedBase.ProxyAdminOwnedBase_NotProxyAdminOrProxyAdminOwner.selector);
-
-        // Call the `upgrade` function with the sender
-        vm.prank(_sender);
-        systemConfig.upgrade(1234, ISuperchainConfig(address(0xdeadbeef)));
-    }
-}
```

Comment:
We're deleting these tests, and will need to dig them up next time we want to add an upgrade function.

A decent way of documenting these tests, and the upgrade() function pattern, might be to create and test a simple contract that `is ProxyAdminOwnedBase, Initializable, ReinitializableBase, ISemver`, which has both an `initialize()` and `upgrade()` function. It wouldn't exactly be reusable as a harness since the arguments vary, but still nice to be able to copy paste when needed.

WDYT?